template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaCloudStorageServiceConfigProvider.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.impala;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil.isVersionNewerOrEqualThanLimited;
+import static com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil.CLOUDERA_STACK_VERSION_7_2_11;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateComponentConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class ImpalaCloudStorageServiceConfigProvider implements CmTemplateComponentConfigProvider {
+
+    private static final String IMPALA_CMD_ARGS_SAFETY_VALVE = "impala_cmd_args_safety_valve";
+
+    private static final String IMPALA_STARTUP_FILESYSTEM_CHECK_DIRS = "--startup_filesystem_check_directories=";
+
+    private static final String HIVE_METASTORE_DIR = "hive.metastore.warehouse.dir";
+
```

Comment:
Import HiveMetastoreCloudStorageServiceConfigProvider.HMS_METASTORE_DIR and HMS_METASTORE_EXTERNAL_DIR

Reply:
Updated